tool/outdate-bundled-gems.rb

```diff
@@ -111,7 +111,7 @@ def each_directory
 
 version = RbConfig::CONFIG['ruby_version']
 curdir.glob(".bundle/{extensions,.timestamp}/#{platform}/*/") do |dir|
-  unless File.basename(dir) == version
+  unless File.basename(dir).start_with?(version)
```

Comment:
So, it looks like, `.bundle/extensions/#{platform}/*/` matches:

1. Building Ruby without --enable-shared: `.bundle/extensions/x86_64-linux/3.2.0+3-static/`
2. Building Ruby with --enable-shared: `.bundle/extensions/x86_64-linux/3.2.0+3/`

On the other hand, the above `RbConfig::CONFIG['ruby_version']` returns the ABI version of baseruby, such as `3.2.0+3`. Note that this doesn't have `-static` whether baseruby is --enable-shared or not.

If you use the same ABI version in baseruby, what happens is (1) runs `curdir.rmdir(dir)` and `make install`  fails, and (2) doesn't run `curdir.rmdir(dir)` and `make install` works. Which means that we want to skip running `curdir.rmdir(dir)` as you did, so this patch looks sane.